src/game/server/neo/neo_message.cpp

```diff
@@ -67,5 +99,29 @@ void CNEO_Message::InputShowMessage(inputdata_t& inputData)
 
 void CNEO_Message::InputHideMessage(inputdata_t& inputData)
 {
-    Q_strncpy(m_NetworkedMessageKey.GetForModify(), "", 255);
+	V_strncpy(m_NetworkedMessageKey.GetForModify(), "", sizeof(m_NetworkedMessageKey));
+	V_strncpy(m_NetworkedSubMessageKey.GetForModify(), "", sizeof(m_NetworkedSubMessageKey));
+	if (m_bTimerMode)
+	{
+		SetNextThink(TICK_NEVER_THINK);
+	}
+}
+
+void CNEO_Message::InputShowSubMessage(inputdata_t& inputData)
+{
+	V_strncpy(m_NetworkedSubMessageKey.GetForModify(), inputData.value.String(), sizeof(m_NetworkedSubMessageKey));
+}
+
+void CNEO_Message::InputHideSubMessage(inputdata_t& inputData)
+{
+	V_strncpy(m_NetworkedSubMessageKey.GetForModify(), "", sizeof(m_NetworkedSubMessageKey));
```

Comment:
Since `strncpy` would zero the entire length of the leftover memory block, this could be optimized to just null-terminate the first character instead: `*m_NetworkedSubMessageKey.GetForModify() = 0;`

This doesn't really matter for performance, but since this is networked, zeroing the whole block might cause a spike in the netcode delta that must be sent to each client, so it'd be nicer to just send the delta of the first element being zeroed since the early null terminator is all we need.